### PR TITLE
astroid: fix python plugin usage

### PR DESCRIPTION
**Pull request by oxzi**
###### Motivation for this change
Astroid's python-based plugin system was not usable before.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuprDewd — APPROVED** [review]
LGTM

File: pkgs/applications/networking/mailreaders/astroid/default.nix [hunk 1]
```diff
@@ -20,22 +20,31 @@ stdenv.mkDerivation rec {
     sha256 = "11cxbva9ni98gii59xmbxh4c6idcg3mg0pgdsp1c3j0yg7ix0lj3";
   };
 
-  nativeBuildInputs = [ cmake ronn pkgconfig wrapGAppsHook ];
+  nativeBuildInputs = [
+    cmake ronn pkgconfig wrapGAppsHook gobject-introspection
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/astroid/default.nix:24]:
```suggestion
    cmake ronn pkg-config wrapGAppsHook gobject-introspection
```

File: pkgs/applications/networking/mailreaders/astroid/default.nix [hunk 2]
```diff
@@ -20,22 +20,31 @@ stdenv.mkDerivation rec {
...
   buildInputs = [
     gtkmm3 gmime3 webkitgtk libsass libpeas
-    python3 python3Packages.pygobject3
+    python3
     notmuch boost gsettings-desktop-schemas gnome3.adwaita-icon-theme
     glib-networking protobuf
-   ] ++ (if vim == null then [] else [ vim ]);
+  ] ++ (if vim == null then [] else [ vim ]);
+
+  enableParallelBuilding = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/astroid/default.nix:35]: cmake does this automatically.

```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please fix the eval error.

File: pkgs/applications/networking/mailreaders/astroid/default.nix [hunk 3]
```diff
@@ -20,22 +20,29 @@ stdenv.mkDerivation rec {
...
+  ];
 
   buildInputs = [
     gtkmm3 gmime3 webkitgtk libsass libpeas
-    python3 python3Packages.pygobject3
+    python3
     notmuch boost gsettings-desktop-schemas gnome3.adwaita-icon-theme
     glib-networking protobuf
-   ] ++ (if vim == null then [] else [ vim ]);
+  ] ++ (if vim == null then [] else [ vim ]);
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/mailreaders/astroid/default.nix:33]:
```suggestion
  ] ++ lib.optionals vim [ vim ]);
```

**Comment by SuperSandro2000** [comment]
@oxzi please fix the merge conflict.

**Comment by oxzi** [comment]
@SuperSandro2000: Thanks, please have another look.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 100642` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>astroid</li>
</ul>
</details>

**Pull request merged**